src/routes/accounts/accounts.service.ts

```diff
@@ -53,6 +56,28 @@ export class AccountsService {
     );
   }
 
+  async upsertAccountDataSettings(args: {
+    authPayload: AuthPayload;
+    address: `0x${string}`;
+    upsertAccountDataSettingsDto: UpsertAccountDataSettingsDto;
+  }): Promise<AccountDataSetting[]> {
+    const domainAccountDataSettings =
+      await this.accountsRepository.upsertAccountDataSettings({
+        authPayload: args.authPayload,
+        address: args.address,
+        upsertAccountDataSettings: {
+          accountDataSettings:
+            args.upsertAccountDataSettingsDto.accountDataSettings,
+        },
+      });
+
+    const dataTypes = await this.accountsRepository.getDataTypes();
```

Comment:
We could wrap these in a `Promise.all`.

Reply:
Done in [2a6e988](https://github.com/safe-global/safe-client-gateway/pull/1715/commits/2a6e988cbd0a6970b9aff2d1efd9b28d43342955)